### PR TITLE
feat(session) forward persistent cookie field

### PR DESCRIPTION
**Pull request by tschaume**
### Summary

This PR forwards the `session.cookie.persistent` field from lua-resty-session. See https://github.com/bungle/lua-resty-session#boolean-sessioncookiepersistent

### Full changelog

* Added `persistent` field to `session.lua` and `schema.lua`

### CODE REVIEW

**Review by bungle — APPROVED** [review]

File: kong/plugins/session/schema.lua [hunk 1]
```diff
@@ -54,6 +54,7 @@ return {
           { cookie_httponly = { type = "boolean", default = true } },
           { cookie_secure = { type = "boolean", default = true } },
           { cookie_discard = { type = "number", default = 10 } },
+          { cookie_persistent = { type = "boolean", default = false } },
```
Comment (mikefero) [inline kong/plugins/session/schema.lua:57]: This is a breaking change for version compatibility with older data planes; updates will need to be made to the [compatibility layer ](https://github.com/Kong/kong/blob/master/kong/clustering/compat/removed_fields.lua)to handle these changes ahead of time of before a release.

**Comment by tschaume** [comment]
@mikefero I added an entry for the `cookie_persistent` field and kong versions older than `2.8.2`. Happy to update if that's not the way to go about it.

**Comment by mikefero** [comment]
> @mikefero I added an entry for the `cookie_persistent` field and kong versions older than `2.8.2`. Happy to update if that's not the way to go about it.

Thanks @tschaume; feel free to revert your commit as there are items in the tests that need to be addressed for the addition into the removed_fields.lua. When this feature lands and is targeted for a particular release the compatibility layer will be updated to handle this new field. Appreciate the effort, it will be easier to address when it gets closer to release time and after this feature has been merged into `master`.

**Comment by tschaume** [comment]
sounds good, thanks @mikefero. Reverted the commit and rebased.

**Comment by hbagdi** [comment]
How does this relate to https://github.com/Kong/kong/pull/8937?

**Comment by tschaume** [comment]
@hbagdi #8937 seems to do the same but adds a test for it. Not sure why a new PR was started for it, though.

**Comment by hbagdi** [comment]
@tschaume Can you rebase, add the relevant test? That will help us merge this.

**Comment by tschaume** [comment]
@hbagdi I rebased and added the tests from #8937

**Review by mayocream — APPROVED** [review]
Missing a changelog entry, but you can add it in a separate PR.

**Comment by dndx** [comment]
@mayocream Could you add the changelog entry for this PR? I think it will be forgotten and not done if merged now.

**Pull request merged**